### ISSUE TITLE
add clarity to subject search feature

### ISSUE THREAD

**Issue by theLinkResolver** [ualbertalib/discovery#864]
Suggestion from training session. The idea is to help users understand what they are doing when they click on the subject headings.

- One idea is to present the subject headings as a hierarchy, with text indented on separate lines, although I think this could get ugly when there are more than two subject headings (quite common).

- Another idea is to have a hover box/tooltip that tells you what is going to happen when you click. Example: https://searchworks.stanford.edu/view/7105806

**Comment by redlibrarian** [comment]
This is one of those areas where I'd rather users spoke for themselves. If this doesn't present itself as a problem either on the desk/in the helpdesk or in usability testing, I think it would be better to leave it be. With regard to the hierarchical display suggestion: it's bad practice to put line breaks in the middle of a hyperlink, though I suppose we could make each line a separate hyperlink which includes the term from the line before, but to my mind that's *more* complicated and unintuitive than what we currently have, because it links out to something not displayed in the hyperlink. We could obviously create a tooltip, but it seems pretty clear that few people read them, and we (the testing and training team) have difficulty reaching a consensus on wording. I don't think think this would be an easy task given the length constraints in a tooltip. 

So, for what it's worth, my suggestion is to wait and see if it turns out that our users actually don't understand what they're doing.

**Comment by theLinkResolver** [comment]
@redlibrarian @gooyers The idea of "find more like this" is a key one, and it would be great to see what happens in future usability testing. e.g. "You have found an excellent book on your topic [link to it] and you want to find others like it."

1. Do users use the subject headings to accomplish this task? (tests the wording, layout, and understanding of the idea of subject headings)
2. If so, do they understand what their clicks are doing, when they click on the heading terms?
3. Are they satisfied with the outcome?

For instance, a book like this: https://www.library.ualberta.ca/catalog/4633129
There could also be a followup question, "if you wanted to see more books on the legal status of women, but not necessarily limited to the United States, how would you find them?"

**Comment by redlibrarian** [comment]
One thing that came up when I was discussing this topic with @psue is that Cornell, where we got the idea for the subject string display/design highlights *all* previous terms in the heading when you click on a particular term, making it clearer to see that your click encompasses all highlighted terms. We aren't doing that, so perhaps it looks as if a click only uses the term actually clicked on. (If that's not clear, see how the highlighting works in https://newcatalog.library.cornell.edu/catalog/8908546). I think this would be an easy enough change to make, and might clear up some confusion.

**Comment by redlibrarian** [comment]
@theLinkResolver @gooyers WRT the "find more like this" functionality, subject headings and other access points (combined with authority records) are how we've traditionally done this. As opposed to systems like bibliocommons which use other methods to determine "more like this-ness", we're still constrained to using our bibliographic data (as opposed to, say, our circulation data) to accomplish this. We might want to have a broader discussion around "finding more like this" within the group, but it might be better for that to be independent of this subject string discussion, if only so that our discussion isn't constrained by this particular technical issue.

**Comment by merisjames** [comment]
I really like how Cornell adds the underline to the earlier subject
segments. It's a minor change, visually, but I think it makes a major
impact, conceptually.

Meris James, MLIS
Public Service Librarian
J.A. Weir Memorial Law Library
meris@ualberta.ca

**Comment by redlibrarian** [comment]
:+1:

**Comment by theLinkResolver** [comment]
Being "constrained to using our bibliographic data" does potentially call into question the value of our authority data, which we invest $$$ in on an ongoing basis. Is this a case of conflicting organizational priorities or just conflicting viewpoints? Even without my cataloguer hat on, I am of the opinion that if we are supporting/investing in authority data, we should ensure that our systems take full advantage of it. (see also #707 and #708 for where we grappling with how authority data is used in the Blacklight search)

**Comment by seanluyk** [comment]
@theLinkResolver  - I totally agree with you. We're sitting on a goldmine of bibliographic data, and our systems need to take full advantage of it if we're to be spending this much public $$$ on it, especially the authority data. That being said, I don't think that iLink ever took full advantage of it either, and Blacklight perhaps gives us an opportunity to better capitalize on our investment. BTW the authority data is extremely useful for music and other specialized kinds of searching, so I think a future phase of implementation should look at these areas a little more closely, and/or the role of authority data in this system more generally.

**Comment by redlibrarian** [comment]
I think we need to clarify what "taking full advantage" means. IMO, all these decisions need to have a public, user-facing benefit. So, for example, maintaining authority *record* within Blacklight doesn't make sense if we can get the same functionality out of the index. So, for example when @seanluyk says "the authority data is extremely useful for music and other specialized kinds of searching", what I would like to see is some concrete examples of how the authority data is leveraged here. Sean's right that iLink doesn't do very much with the authority data (though, again, I think "full advantage" is still undefined). Early automation ideas were that authority data would be leveraged for autocorrection/suggestion of search terms, but I've never seen a system that does this. Currently, as far as I can tell, OPACs generally only use the authority records to allow users to click on authors, subjects, uniform titles, etc, in order to get all records "attached" to that heading. Blacklight achieves the same thing with faceting, so the authority records aren't required (though the consistent labeling we get from cataloguing *with* the authority records is required). It's important to remember that the authority structure was initially designed for a card-catalogue world, and that thinking is still prevalent in ILSs and OPACs. 

tl;dr what are the concrete use-cases we think we can leverage authority data to achieve?

**Comment by theLinkResolver** [comment]
For me in this context, my bar isn't too high -- "taking full advantage" is ensuring all subjects & names are clickable, that clicking has the expected results (more like this/by this author with high precision), and most importantly, the option to do this is presented in a way that gives people a reason to click them and reasonably understand what is going to happen. (In this sense, we are already doing something post-OPAC with the ability to click within a subject string, as long as people understand why they would want to click them at all.) My sentiment is that the perception of headings as constraining and relics of an antiquated (card-catalogue) world *might* be out of step with other actively pursued initiatives/goals -- and vice-versa.

**Comment by redlibrarian** [comment]
OK, fair enough. We don't need "authority data" in order to do these things - though the author and subject strings need to conform to a single authoritative version in order for this kind of access to work. The other side of this coin (presentation) has, IMO, less to do with data than with UX (though of course, both are involved). 

"My sentiment is that the perception of headings as constraining and relics of an antiquated (card-catalogue) world might be out of step with other actively pursued initiatives/goals -- and vice-versa." yes, I'd agree with that - this is the high-level perspective I think we should be tackling.

**Comment by seanluyk** [comment]
I completely agree @redlibrarian, and for me, "taking full advantage" **always** implies a public, user-facing benefit, as witnessed through direct experiences working with users.

There are a few specific ways that I think authority data can be used that the index currently doesn't accomplish. These are music specific examples, but perhaps they could be extended to other use cases:

*Viewing an ordered list of the works of a composer (or author)*
**Use Case**: I want to view a list (by title, form, instruments, key) of the scores of a specific composer available at Rutherford library, so that I can choose repertoire for a concert, and/or write a paper about the stylistic development of of this composer's works.
- This was easily achievable in a card catalogue, and never quite made it to the OPAC in a useful way
- iLink currently allows you to do an author starting with search to browse author headings combined with either the uniform title, and/or 700 $t fields, and perhaps more(?). This is admitedly clunky for a number of reasons (no way to simultaneously limit by format or location, and librarian probably needs to teach a user to do this as it isn't intuitive), but it at least allows someone to get to all versions of a work in a (mostly) reliable way, and serves as a way to browse by author and title combined, rather than having to know in advance what controlled vocabulary needs to be included in a search
- Authority data for these fields also have very useful cross-references, so that the user retrieves all version of the same work as they are filed under an authorized heading. This takes care of variant titles ('popular' titles, different language of publication, etc.), and different manifestations of the same work (e.g. a piano reduction of an opera, just the 4th movement of a symphony, not the whole work)
**How authority data could be used in Blacklight to achieve this?**: The Wisconsin browse-search could certainly be used for this purpose, and combined with faceting and powerful relevance ranking, would be superior. Even if uniform titles and 700 fields were hyperlinked, we could probably get close to this functionality. If faceting could achieve this (e.g. if uniform titles could be split into the separate facets of form, instruments, numbers, and key as this field is constructed) that would be even better

*Finding all scores or recordings of a musical work*
**Use Case**: I want to view a list of all the works of a composer or performances by a particular performed by title, so that I can view all versions of a work to pick the best one, and/or listen to all performances of a particular work by various performers

- This is currently not possible in Blacklight, although we do get very very close through faceting, but miss items because authority data isn't currently utilized, and include items that shouldn't be included (if the user task is to find only the items they are looking for, not discover close matches)
- For example, a user wants to find all versions in our collection of Stravinsky's the Rite of Spring
- An author-title starting with browse in iLink for Stravinsky, Igor, 1882-1971. Rite of spring leads to a cross reference for Stravinsky, Igor, 1882-1971. Vesna svi͡ashchennai͡a, which returns 123 results (all formats). But if I search for Vesna svi͡ashchennai͡a in any field, and limit to type = music score, I get 15 results   
- A keyword search (for Stravinsky Rite of Spring) in Blacklight returns 269 results in Books, Media, and More
-Faceting by author and format to Stravinsky and Musical Score provides 17 results, but it misses some items that iLink provides ([this one](http://www.library.ualberta.ca/permalink/opac/6453160/WUAARCHIVE) and [this one](http://www.library.ualberta.ca/permalink/opac/495150/WUAARCHIVE), and includes ones that aren't this musical work ([this is just one movement](https://search.library.ualberta.ca/catalog/5398848), and [this is not this work](https://search.library.ualberta.ca/catalog/195675)

**How authority data could be used in Blacklight to achieve this?**: I'm not really sure how authority data would help, other than some sort of browse search as mentioned above, or perhaps "did you mean" functionality. If Rite and Spring and Stravinsky aren't in a record, Blacklight isn't going to return it. We could look at this as a data problem (which it is), but we've got to work with the catalogue records we've got, too

#707 is also a really important issue that I won't duplicate here. That data should at least be indexed, as it contains information often not included in a uniform title, that's essential for discovery.

These are just two examples, the Music Discovery Requirements (section IVb) explain this issue much better than me. I'll sumarize their recommendations in another post to this issue. 

Blacklight currently does not accommodate information seeking methods that are not at either extremes of known-item searching, and broad based discovery. I'm not sure if using authority data for this purpose is the answer, but I think it's something we need to continue to work on.

**Comment by redlibrarian** [comment]
Thanks @seanluyk - these are great. There's a lot to think about here. @ualbertalib/discovery-testing if anyone has other use-cases that haven't been talked about, feel free to add them to this issue.

**Comment by seanluyk** [comment]
Thanks @redlibrarian, and you should know that overall I'm very happy with music search in Blacklight (it's much improved over all of our previous discovery systems). Music students were able to successfully find scores and recordings very easily in a library workshop I provided last week, and faceting did most of the work for them.

**Comment by seanluyk** [comment]
Long and short of the [Music Discovery Requirements](http://committees.musiclibraryassoc.org/ETSC/MDR) as they pertain to authority records:
-Creators are important access points for music search, and names vary. Works associated with more than one creator (composer, performer, librettist, etc.) also present challenges. Methods should be provided to lead users from alternate names to the ones they seek. I think this use case applies not just to music, and especially to other fields in the arts (esp. drama, literature)
-Linked data via 7xx fields could open up many more possibilities for discovery, for example, taking advantage of autocomplete suggestions, and users being able to search in their native language, etc.
-Other ways of manipulating elements that identify works (e.g. viewing a composer's works by opus number) should be possible in the future, and I think are quite desirable

Main Recommendations:
"For public interfaces, browse indexes should display 4xx and 5xx cross-references and
public notes such as the 680. Associated name and title strings must be kept together for both indexing and display...Ideally, as discovery systems evolve, they should be able to index and display cross-references in keyword indexes and allow linking or display of alternate data. They should leverage authority information to provide autosuggestions, context-sensitive recommendations, or other functionality. Additionally, they should be able to make use of extended authority fields for display or faceting."
